### PR TITLE
Remove some trivial private getters/setters in axisartist

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

- Remove trivial private `_get_offset_radius` / `_set_offset_radius`
- Remove trivial private `_get_ref_angle` / `_set_ref_angle`
- Remove trivial private `_get_external_pad` / `_set_external_pad`
- Turn `_get_text_ref_angle` and `_get_offset_ref_angle` into properties

### CODE REVIEW

File: lib/mpl_toolkits/axisartist/axis_artist.py [hunk 1]
```diff
@@ -378,19 +356,17 @@ def draw(self, renderer):
         if not self.get_visible():
             return
 
-        pad = renderer.points_to_pixels(self.get_pad())
-        r = self._get_external_pad() + pad
-        self._set_offset_radius(r)
+        r = self._external_pad + renderer.points_to_pixels(self.get_pad())
```
Comment (anntzer) [inline lib/mpl_toolkits/axisartist/axis_artist.py:359]: You can even get rid of the `r` intermediate variable (below too).

File: lib/mpl_toolkits/axisartist/axis_artist.py [hunk 2]
```diff
@@ -541,20 +516,19 @@ def set_locs_angles_labels(self, locs_angles_labels):
...
 
         bboxes = []
 
         r, total_width = self._get_ticklabels_offsets(renderer,
                                                       self._axis_direction)
 
-        pad = self._get_external_pad() + \
-            renderer.points_to_pixels(self.get_pad())
-        self._set_offset_radius(r+pad)
+        pad = self._external_pad + Trenderer.points_to_pixels(self.get_pad())
```
Comment (anntzer) [inline lib/mpl_toolkits/axisartist/axis_artist.py:527]: typo here.

**Review by anntzer — APPROVED** [review]
modulo CI.

File: lib/mpl_toolkits/axisartist/axis_artist.py [hunk 3]
```diff
@@ -249,14 +239,12 @@ def draw(self, renderer):
...
-        offset_ref_angle = self._get_offset_ref_angle()
-        theta = np.deg2rad(offset_ref_angle)
-        dd = self._get_offset_radius()
+        theta = np.deg2rad(self._offset_ref_angle)
+        dd = self._offset_radius
         dx, dy = dd * np.cos(theta), dd * np.sin(theta)
 
         self.set_transform(tr + Affine2D().translate(dx, dy))
-        self.set_rotation(text_ref_angle+angle_orig)
+        self.set_rotation(self.text_ref_angle + angle_orig)
```
Comment (QuLogic) [inline lib/mpl_toolkits/axisartist/axis_artist.py:247]:
```suggestion
        self.set_rotation(self._text_ref_angle + angle_orig)
```

File: lib/mpl_toolkits/axisartist/axis_artist.py [hunk 4]
```diff
@@ -266,14 +254,12 @@ def get_window_extent(self, renderer):
...
```
Comment (QuLogic) [inline lib/mpl_toolkits/axisartist/axis_artist.py:262]:
```suggestion
        self.set_rotation(self._text_ref_angle + angle_orig)
```

**Review by QuLogic — COMMENTED** [review]
At least these, based on tests.

**Review by greglucas — APPROVED** [review]

**Pull request merged**